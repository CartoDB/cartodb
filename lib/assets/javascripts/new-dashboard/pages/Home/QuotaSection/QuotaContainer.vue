<template>
  <div class="quota-container">
    <div class="quota-headers">
      <div class="quota-cell cell--main">
        <h3 class="title is-caption quota-title">{{title}}</h3>
      </div>

      <div class="quota-labels">
        <div class="quota-cell cell--medium">
          <span class="text is-small is-txtSoftGrey">{{ $t('QuotaSection.header.remaining') }}</span>
        </div>
        <div class="quota-cell cell--medium cell--mobile">
          <span class="text is-small is-txtSoftGrey">{{ $t('QuotaSection.header.used') }}</span>
        </div>
        <div class="quota-cell cell--medium">
          <span class="text is-small is-txtSoftGrey quota-total" :class="{'is-active' : perMonth }">{{$t(`QuotaSection.header.total`)}}</span>
          <span class="text is-small is-txtSoftGrey quota-month" :class="{'is-active' : perMonth }">&nbsp;{{$t(`QuotaSection.header.perMonth`)}}</span>
        </div>
        <div class="quota-cell cell--small"></div>
      </div>
    </div>
    <div class="quota-info">
      <slot></slot>
    </div>
  </div>
</template>

<script>
export default {
  name: 'QuotaContainer',
  props: {
    title: String,
    perMonth: Boolean
  }
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.quota-headers {
  display: flex;
  justify-content: space-between;
}

.quota-title {
  margin-bottom: 8px;
}

.quota-labels {
  display: flex;
}

.quota-info {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: $white;
}

.quota-cell {
  display: flex;
  align-items: center;
  padding: 0 10px;
  white-space: nowrap;

  &:first-of-type {
    padding-left: 0;
  }

  &:last-of-type {
    padding-right: 0;
  }
}

.cell--mobile {
  @media (max-width: $layout-mobile) {
    display: none;
  }
}

.cell--main {
  min-width: 300px;

  @media (max-width: $layout-mobile) {
    min-width: unset;
  }
}

.cell--medium {
  width: 120px;
}

.cell--small {
  width: 58px;

  @media (max-width: $layout-tablet) {
    display: none;
  }
}

.quota-month {
  display: none;
}

.is-active {
  &.quota-total {
    @media (max-width: $layout-tablet) and (min-width: $layout-mobile) {
      &::after {
        content: '*';
      }
    }
  }

  &.quota-month {
    display: flex;

    @media (max-width: $layout-tablet) and (min-width: $layout-mobile) {
      display: none;
    }

    &::after {
      content: '*';
    }
  }
}

</style>
