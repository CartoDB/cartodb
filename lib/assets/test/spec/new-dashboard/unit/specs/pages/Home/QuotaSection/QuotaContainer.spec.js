import { shallowMount } from '@vue/test-utils';
import QuotaContainer from 'new-dashboard/pages/Home/QuotaSection/QuotaContainer';

const $t = key => key;

describe('QuotaContainer.vue', () => {
  it('should render correct container with headers for quotas including total per month', () => {
    const quotaContainer = shallowMount(QuotaContainer, {
      propsData: { title: 'Test Title', perMonth: true },
      mocks: {
        $t
      }
    });

    expect(quotaContainer).toMatchSnapshot();
  });
});
