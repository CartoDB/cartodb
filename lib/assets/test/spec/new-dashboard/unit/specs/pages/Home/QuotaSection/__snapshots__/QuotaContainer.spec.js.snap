// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuotaContainer.vue should render correct container with headers for quotas including total per month 1`] = `
<div class="quota-container">
  <div class="quota-headers">
    <div class="quota-cell cell--main">
      <h3 class="title is-caption quota-title">Test Title</h3>
    </div>
    <div class="quota-labels">
      <div class="quota-cell cell--medium"><span class="text is-small is-txtSoftGrey">QuotaSection.header.remaining</span></div>
      <div class="quota-cell cell--medium cell--mobile"><span class="text is-small is-txtSoftGrey">QuotaSection.header.used</span></div>
      <div class="quota-cell cell--medium"><span class="text is-small is-txtSoftGrey quota-total is-active">QuotaSection.header.total</span> <span class="text is-small is-txtSoftGrey quota-month is-active">&nbsp;QuotaSection.header.perMonth</span></div>
      <div class="quota-cell cell--small"></div>
    </div>
  </div>
  <div class="quota-info"></div>
</div>
`;
